isis/src/system/objs/KernelDb/KernelDb.cpp

```diff
@@ -586,10 +586,10 @@ namespace Isis {
     QString instrument = "";
 
     if (grp.hasKeyword("StartOffset")){
-      startOffset = (double) grp["StartOffset"];
+      startOffset = (double) grp["StartOffset"] + 0.001;
     }
     if (grp.hasKeyword("EndOffset")){
-      endOffset = (double) grp["EndOffset"];
+      endOffset = (double) grp["EndOffset"] + 0.001;
```

Comment:
Running into an issue that these offsets where being added to kernel times with seconds that go out to the millionth decimal place (e.g. 59.720231) and the offsets and label times were only to the thousandth decimal place (e.g. 57.253). Therefore after adding an offset to the kernel time, it would still have an additional 0.000xxx number causing the comparison to fail. Adding 0.001 handles this. I could not find a better way to go about this since you cannot round iTime seconds.